simplefs/simplefs.go

```diff
@@ -386,15 +612,34 @@ func (k *SimpleFS) SimpleFSRename(ctx context.Context, arg keybase1.SimpleFSRena
 	}
 	defer func() { k.doneSyncOp(ctx, err) }()
 
-	snode, sleaf, err := k.getRemoteNodeParent(ctx, arg.Src)
+	// Get root FS, to be shared by both src and dst.
+	t, tlfName, restOfSrcPath, finalSrcElem, err := remotePath(arg.Src)
+	if err != nil {
+		return err
+	}
+	tlfHandle, err := libkbfs.GetHandleFromFolderNameAndType(
+		ctx, k.config.KBPKI(), k.config.MDOps(), tlfName, t)
 	if err != nil {
 		return err
 	}
-	dnode, dleaf, err := k.getRemoteNodeParent(ctx, arg.Dest)
+	fs, err := libfs.NewFS(
+		ctx, k.config, tlfHandle, "", "", keybase1.MDPriorityNormal)
 	if err != nil {
 		return err
 	}
-	err = k.config.KBFSOps().Rename(ctx, snode, sleaf, dnode, dleaf)
+
+	// Make sure src and dst share the same TLF.
+	tDst, tlfNameDst, restOfDstPath, finalDstElem, err := remotePath(arg.Dest)
+	if err != nil {
+		return err
+	}
+	if tDst != t || tlfName != tlfNameDst {
+		return simpleFSError{"Cannot rename across top-level folders"}
+	}
+
+	err = fs.Rename(
+		fs.Join(restOfSrcPath, finalSrcElem),
```

Comment:
I guess using `fs.Join` is more portable, since technically it's the `FS`'s responsibility to determine how paths can be joined. But in a lot if other places in this file we are using `stdpath.Join`. They mean the same thing but it'd be nice to unify them.